bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpClientFacade.java

```diff
@@ -0,0 +1,156 @@
+
+    /**
+     * Performs a POST request to the specified URI with the given form parameters.
+     *
+     * @param uri the URI to POST
+     * @param fields the fields composing the form name/value pairs
+     * @return the {@link ContentResponse} for the request
+     * @throws InterruptedException if send threading has been interrupted
+     * @throws ExecutionException the execution failed
+     * @throws TimeoutException the send timed out
+     */
+    ContentResponse FORM(String uri, Fields fields) throws InterruptedException, ExecutionException, TimeoutException;
+
+    /**
+     * Performs a POST request to the specified URI with the given form parameters.
+     *
+     * @param uri the URI to POST
+     * @param fields the fields composing the form name/value pairs
+     * @return the {@link ContentResponse} for the request
+     * @throws InterruptedException if send threading has been interrupted
+     * @throws ExecutionException the execution failed
+     * @throws TimeoutException the send timed out
+     */
+    ContentResponse FORM(URI uri, Fields fields) throws InterruptedException, ExecutionException, TimeoutException;
+
+    /**
+     * Creates a POST request to the specified URI.
+     *
+     * @param uri the URI to POST to
+     * @return the POST request
+     * @see #POST(URI)
+     */
+    Request POST(String uri);
+
+    /**
+     * Creates a POST request to the specified URI.
+     *
+     * @param uri the URI to POST to
+     * @return the POST request
+     */
+    Request POST(URI uri);
+
+    /**
+     * Creates a new request with the "http" scheme and the specified host and port
+     *
+     * @param host the request host
+     * @param port the request port
+     * @return the request just created
+     */
+    Request newRequest(String host, int port);
+
+    /**
+     * Creates a new request with the specified absolute URI in string format.
+     *
+     * @param uri the request absolute URI
+     * @return the request just created
+     */
+    Request newRequest(String uri);
+
+    /**
+     * Creates a new request with the specified absolute URI.
+     *
+     * @param uri the request absolute URI
+     * @return the request just created
+     */
+    Request newRequest(URI uri);
+
+    /**
+     * Returns a {@link Destination} for the given scheme, host and port.
+     * Applications may use {@link Destination}s to create {@link Connection}s
+     * that will be outside {@link HttpClient}'s pooling mechanism, to explicitly
+     * control the connection lifecycle (in particular their termination with
+     * {@link Connection#close()}).
+     *
+     * @param scheme the destination scheme
+     * @param host the destination host
+     * @param port the destination port
+     * @return the destination
+     * @see #getDestinations()
+     */
+    Destination getDestination(String scheme, String host, int port);
+
+    boolean isDefaultPort(String scheme, int port);
```

Comment:
I know that it looks as same as JettyClient's HTTP Client. It would be really helpful for consumers to have basic javadoc of this method's contract.

Reply:
There is no javadoc in the original jetty http client for this method as well.

Reply:
I have seen that and that is why I thought we could add something small. 

For example,

```java
/**
 * Checks if the provided URI scheme associates the provided port
 *  
 * @param scheme URI scheme
 * @param port port number to check
 *
 * @throws NullPointerException if URI scheme is {@code null}
 */
```

Reply:
No. I will no provide a documentation based on assumptions based on the method name. That does not help anyone.

Reply:
Justified 👍